Branch-SDK/Branch-SDK/BNCSystemObserver.m

```diff
@@ -121,13 +121,16 @@ + (NSString *)getDeviceName {
 }
 
 + (NSNumber *)getUpdateState {
-    NSString *bundleRoot = [[NSBundle mainBundle] bundlePath];    NSFileManager *manager = [NSFileManager defaultManager];
-    NSDictionary* attrs = [manager attributesOfItemAtPath:bundleRoot error:nil];
-    if ((int)([[attrs fileCreationDate] timeIntervalSince1970]/(60*60*24)) == (int)([[attrs fileModificationDate] timeIntervalSince1970]/(60*60*24))) {
-        return nil;
-    } else {
+
+    NSString *storedAppVersion = [BNCPreferenceHelper getAppVersion];
+    NSString *currentAppVersion = [BNCSystemObserver getAppVersion];
+
+    if (storedAppVersion == nil ||
+        ([storedAppVersion compare:currentAppVersion options:NSNumericSearch] == NSOrderedAscending)) {
+        [BNCPreferenceHelper setAppVersion:currentAppVersion];
```

Comment:
What's the difference between scenario two and three there?

Also, wouldn't it just be easiest to say `if (![storedAppVersion equal:currentAppVersion])`, since basically any time they aren't the same, we need an update?